components/script/script_thread.rs

```diff
@@ -477,33 +510,23 @@ impl ScriptThread {
         })
     }
 
-    pub(crate) fn set_mutation_observer_microtask_queued(value: bool) {
-        with_script_thread(|script_thread| {
-            script_thread.mutation_observer_microtask_queued.set(value);
-        })
-    }
-
-    pub(crate) fn is_mutation_observer_microtask_queued() -> bool {
-        with_script_thread(|script_thread| script_thread.mutation_observer_microtask_queued.get())
-    }
-
     pub(crate) fn add_mutation_observer(observer: &MutationObserver) {
         with_script_thread(|script_thread| {
             script_thread
+                .mutation_observers
                 .mutation_observers
                 .borrow_mut()
                 .push(Dom::from_ref(observer));
         })
     }
 
-    pub(crate) fn get_mutation_observers() -> Vec<DomRoot<MutationObserver>> {
-        with_script_thread(|script_thread| {
-            script_thread
+    pub(crate) fn mutation_observers() -> MutationObservers {
+        with_script_thread(|script_thread| MutationObservers {
+            mutation_observer_microtask_queued: script_thread
                 .mutation_observers
-                .borrow()
-                .iter()
-                .map(|o| DomRoot::from_ref(&**o))
-                .collect()
+                .mutation_observer_microtask_queued
+                .clone(),
+            mutation_observers: script_thread.mutation_observers.mutation_observers.clone(),
```

Comment:
Cloning this list means we won't affect the original. We need to return a `Rc<MutationObservers>` from this method instead, and store that inside ScriptThread.